R/sample.R

```diff
@@ -0,0 +1,89 @@
+#' **DESCRIPTION**
+#'
+#' \code{sample} **DESCRIPTION**
+#'
+#' @param batch_id        Batch ID.
+#' @param pattern         Regular expression - only CSVs.
+#' @param output          Output file - either CSV or RDS.
+#' @param replicates      Number of replicates to select per plate map. Optional. default: \code{NULL}.
+#' @param workspace_dir   Root directory containing backend and metadata subdirectories. Can be relative or absolute. default: \code{"."}.
+#' @importFrom magrittr %<>%
+#' @importFrom magrittr %>%
+#' @importFrom magrittr extract2
+#' @importFrom rlang .data
+#' @importFrom utils head
+#' @importFrom utils tail
+#' @export
+sample <- function(batch_id, pattern, output,
+                   replicates = NULL,
+                   workspace_dir = ".") {
+
+  backend_dir <- paste(workspace_dir, "backend", batch_id, sep = "/")
+
+  metadata_dir <- paste(workspace_dir, "metadata", batch_id, sep = "/")
+
+  file_list <- list.files(backend_dir,
+                          pattern = pattern,
+                          recursive = T, full.names = T)
+
+  if (!is.null(replicates)) {
+    # get the list of plates that retrieved using the pattern
+    plate_list_retrieved <-
+      dplyr::data_frame(
+        Assay_Plate_Barcode = lapply(file_list, function(file) {
+          head(tail(stringr::str_split(file, "/")[[1]], 2), 1)
+        }) %>%
+          unlist()
+        )
+
+    replicates <- as.integer(replicates)
+
+    # create a plate_list based on number of replicates to be selected
+    plate_list <-
+      suppressMessages(
+        readr::read_csv(
+          paste(metadata_dir, "barcode_platemap.csv", sep = "/"),
+          col_types = readr::cols(Assay_Plate_Barcode = readr::col_character(),
+                                  Plate_Map_Name = readr::col_character()))) %>%
+      dplyr::select("Assay_Plate_Barcode", "Plate_Map_Name") %>%
+      dplyr::inner_join(plate_list_retrieved, by = "Assay_Plate_Barcode") %>%
+      dplyr::group_by("Plate_Map_Name") %>%
+      dplyr::arrange(Assay_Plate_Barcode) %>%
```

Comment:
sample: no visible binding for global variable ‘Assay_Plate_Barcode’

Reply:
I'm unsure how to resolve this as the documentation states the arguments should be unquoted variable names:

> .data A tbl. All main verbs are S3 generics and provide methods for tbl_df(), dtplyr::tbl_dt() and dbplyr::tbl_dbi().
> ...  **Comma separated list of unquoted variable names.** Use desc() to sort a variable in descending order.
> .by_group If TRUE, will sort first by grouping variable. Applies to grouped data frames only.